python/lsst/cp/pipe/defects.py

```diff
@@ -246,7 +247,15 @@ def findHotAndColdPixels(self, exp, nSigma):
 
                 threshold = afwDetection.createThreshold(nSig, 'stdev', polarity=polarity)
 
-                footprintSet = afwDetection.FootprintSet(ampImg, threshold)
+                try:
+                    footprintSet = afwDetection.FootprintSet(ampImg, threshold)
+                except InvalidParameterError:
+                    # This occurs if the image sigma value is 0.0.
+                    # Let's mask the whole area.
+                    minValue = np.nanmin(ampImg.image.array) - 1.0
+                    threshold = afwDetection.createThreshold(minValue, 'value', polarity=True)
+                    footprintSet = afwDetection.FootprintSet(ampImg, threshold)
```

Comment:
This seems like a pretty complicated (and expensive) way to create a `FootprintSet` covering the whole area.  Since a `FootprintSet` can be created from a `Box2I`, wouldn't that be simpler and faster?

Reply:
I think I got lost in the `FootprintSet` definition before I saw that it could have a `Box` passed to it.  I'll refactor for that instead.

Reply:
After coding this and sending to Jenkins, I'm now discovering that the box definition only sets the region, and does not create a footprint set containing footprints that can be used to mask the image.